lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
It will be more to type for the end-user if that is relevant.  Also, that will need a change in spack to do smth different for the XL compiler (which may already do, but it doesn't seem to fit into the model, where the compilers definition is isolated in their respective classes.)  IMO, it is cleaner to just define a new compiler configuration that uses all _r stanzas.